pkg/objstore/client/factory.go

```diff
@@ -41,6 +41,7 @@ const (
 type BucketConfig struct {
 	Type   ObjProvider `yaml:"type"`
 	Config interface{} `yaml:"config"`
+	Prefix string      `yaml:"prefix" default:""`
```

Comment:
> Yeah, we generate those docs automatically from the structs. Since this option is "hidden" now in the types, the check cannot pass. What about adding a new field to the [BucketConfig type](https://github.com/thanos-io/thanos/blob/7ba274c69cb0a848c7e8e1c593221faa80d8b6e6/pkg/objstore/client/factory.go#L41)? Then we wouldn't have to have hacks like [this](https://github.com/thanos-io/thanos/pull/5337/files#diff-aba15f20dff65b99f7320cb411ad588f2ee051714d3bda4d986b5a05d85bb105R91).

Hey @GiedriusS , I followed your suggestion here :) The only thing I don't love about bringing the prefix to the BucketConfig is that all the bucket options are inside Config 🤔
But with this change, we also keep the prefix layer with its validations and don't  take the risk to break something by changing all the Config types that BucketConf handles, so I think it's the best alternative

Reply:
:+1: yeah, I think this is the best solution out of the alternatives